### PR TITLE
[UNIONVMS-4297]Trip summary refactoring and associated tests

### PR DESCRIPTION
**Pull request by cmakri**
implementation and unit tests. Please check css for pies @ trip summary

### CODE REVIEW

File: app/directive/activity/catchPanel/catchPanel.html [hunk 1]
```diff
@@ -2,54 +2,45 @@
     <div class="row">
         <fieldset class="col-md-12 fieldsetStyle catch-section">
             <legend class="fieldsetLegendStyle">
-                <a>{{title}}</a>
+                <span class="legend-title">
+                    Catch Summary
```
Comment (nikospara) [inline app/directive/activity/catchPanel/catchPanel.html:6]: I think we should continue using `{{title}}` here. If we need a different wording, let's change it in the i18n files.

File: app/directive/activity/catchPanel/catchPanel.html [hunk 2]
```diff
@@ -2,54 +2,45 @@
...
-                                                <div class="td col-md-6 column-right" title="{{item.weight}}">{{item.weight | number:2}}</div>
-                                            </div>
-                                        </div>
+            <div class="row catch-details-body">
+                <div class="col-md-12 col-lg-12">
+                    <div class="col-sm-8 col-md-8 col-xs-12 catch-details-item summary-details-pie">
+                        <div class="{{ngModel.graphClass}} summary-pie-section" ng-repeat="item in ngModel.tripCatchSummary track by $index" ng-if="item.onBoard.length || item.transhipment.length ||item.landing.length">
+                            <fieldset class="summary-section catch-details-item-fieldsetStyle chart-section">
+                                    <legend class="fieldsetLegendStyle summary-pie-section-title">
+                                        <span class="onboard-section-title" ng-if="item.onBoard.length">ONBOARD</span>
```
Comment (nikospara) [inline app/directive/activity/catchPanel/catchPanel.html:15]: Please use `i18n`!

File: app/directive/activity/catchPanel/catchPanel.less [hunk 3]
```diff
@@ -92,4 +94,26 @@ copy of the GNU General Public License along with the IFDM Suite. If not, see <h
             }
         }
     }
+
+    .catch-details-item-fieldsetStyle {
+        border: 1px solid #e5e5e5;
```
Comment (nikospara) [inline app/directive/activity/catchPanel/catchPanel.less:99]: Are there LESS variables for these colors?

File: app/partial/spatial/tripsPanel/tripModel.js [hunk 4]
```diff
@@ -26,7 +26,7 @@ copy of the GNU General Public License along with the IFDM Suite. If not, see <h
  * @description
  *  A model to store all the data related to a trip
  */
-angular.module('unionvmsWeb').factory('Trip',function(locale,unitConversionService,tripReportsTimeline,fishingActivityService) {
+angular.module('unionvmsWeb').factory('Trip',function($http,locale,unitConversionService,tripReportsTimeline,fishingActivityService, tripDataHelperService) {
```
Comment (nikospara) [inline app/partial/spatial/tripsPanel/tripModel.js:29]: Do we need the `$http` here?

File: app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+/**
+ * @memberof unionvmsWeb
+ * @ngdoc service
+ * @name tripDataHelperService
+ * @param $resource {service} angular resource service
+ * @param $http {service} angular http service 
+ * @description
+*/
+
+(function() {
```
Comment (nikospara) [inline app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js:21]: The IIFE won't make any difference here! Everything is `angular.module(...).service(---> HERE <---)`.

File: app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+            this.calculateTableTotals();
+            this.calculateDifferences();
+            self.tripCatchSummary = catchSummary;
+            self.tripTableData  = tableData;
+            self.tripTableHeaders = tableHeaders;
+    
+            return self;
+        },
+    
+        calculateCatchesSummary: function(data){
```
Comment (nikospara) [inline app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js:80]: Does the service need to expose this method in its public API?

I think you can remove most of the methods below from the public API!

File: app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+    angular.module('unionvmsWeb').service('tripDataHelperService', function () {
+        var startingActivityTypes = ['DEPARTURE', 'AREA_ENTRY', 'AREA_EXIT'];
+        var operationActivityTypes = ['FISHING_OPERATION', 'DISCARD', 'RELOCATION', 'JOINED_FISHING_OPERATION'];
+        var speciesColors = [];
+    
+        // TRANSFORMATIONS FOR TABLE
+        var TOTAL = 'TOTAL';
+        var SPECIES = 'Species';
+        var DIFFERENCE = 'Difference';
+        var tableHeaders = [];
```
Comment (nikospara) [inline app/partial/spatial/tripsPanel/tripSummary/tripDataHelperService.js:31]: I feel `tableHeaders`, `tableData`, `dataForTable` should be locals - let's discuss.

**Comment by cmakri** [comment]
should add tests for catch-panel-table too

**Pull request closed without merging**